### PR TITLE
EIP-145: Add test cases

### PR DESCRIPTION
**Pull request by chfast**
I made them by hand, so please check carefully.

### CODE REVIEW

**Comment by axic** [comment]
I think this is missing the case of overflow shifting non-negative numbers with `SAR`, such as `0x7ff..ff << 257` results in `0`.

Also should have tests for the rounding differences:
> This is not equivalent to `PUSH1 2 EXP SDIV`, since it rounds differently. See `SDIV(-1, 2) == 0`, while `SAR(-1, 1) == -1`.

**Comment by chfast** [comment]
`SAR(-1, 1) == -1` is:
```
PUSH 0xffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff
PUSH 0x01
SAR
---
0xffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff
```

**Comment by chfast** [comment]
Is this ok now?

File: EIPS/eip-145.md [hunk 1]
```diff
@@ -79,7 +79,282 @@ The newly introduced instructions have no effect on bytecode created in the past
...
+   SAR
+   ---
+   0x0000000000000000000000000000000000000000000000000000000000000000
+   ```
+3. ```
+   PUSH 0x8000000000000000000000000000000000000000000000000000000000000000
+   PUSH 0x01
+   SAR
+   ---
+   0x12000000000000000000000000000000000000000000000000000000000000000
```
Comment (pirapira) [inline EIPS/eip-145.md]: I guess this shouldn't be like this.  The most significant bit turned from 1 to 0.

Comment (chfast) [reply EIPS/eip-145.md]: oopsy. It should be 12 hex, i.e. 0xc

**Pull request merged**